s3pypi/core.py

```diff
@@ -68,8 +68,10 @@ def upload_packages(
                     index.filenames[filename] = Hash.of("sha256", distr.local_path)
 
     if put_root_index:
-        with storage.locked_index(storage.root) as root_index:
-            root_index.filenames = dict.fromkeys(storage.list_directories())
+        directories = storage.list_directories()
+        if directory not in directories:
```

Comment:
Won't this always be true? `storage.list_directories()` lists the contents of S3, not the contents of the index.

Reply:
I had a closer look, and found that the tests are passing for the wrong reasons.

```python
> /Users/matteo/gorilla/s3pypi-ethanhs/s3pypi/core.py(73)upload_packages()
-> if directory not in directories:
(Pdb) p directory
'xyz'
(Pdb) p directories
['foo/', 'hello-world/', 'xyz/']
```

Reply:
Yeah, I don't really know what I was thinking last night sorry. I should get the list of directories *before* uploading the wheel, as then we will know if the directory the wheel resides in was added or not.

Reply:
I also want to add a test to make sure this works correctly.